base/shell/explorer/taskswnd.cpp

```diff
@@ -1645,6 +2236,89 @@ class CTaskSwitchWnd :
         return FALSE;
     }
 
+    HRESULT CloseOpenedTaskGroup(BOOL bDontRemoveIndices)
+    {
+        if (shellMenu == NULL || menuPopup == NULL || bandSite == NULL || TaskGroupOpened < 0)
+            return E_FAIL;
+
+        PTASK_GROUP TaskGroup = FindTaskGroupByIndex(TaskGroupOpened);
+        if (TaskGroup == NULL)
+            return E_FAIL;
+
+        /*shellMenu->SetSubMenu(menuPopup, FALSE);
+
+        // remove the popup
+        shellMenu = NULL;
+        menuPopup = NULL;
+        bandSite = NULL;*/
+
+        if (!bDontRemoveIndices)
+        {
+            PTASK_ITEM TaskItem, LastTaskItem = NULL;
+
+            TaskItem = m_TaskItems;
+            LastTaskItem = TaskItem + m_TaskItemCount;
+            if (TaskGroupOpened != -1)
+            {
+                while (TaskItem != LastTaskItem)
+                {
+                    if (TaskItem->Group == TaskGroup)
+                    {
+                        TaskItem->GroupIndex = -1;
+                    }
+
+                    TaskItem++;
+                }
+            }
+
+            // Avoid a stack overflow
```

Comment:
Why would a stack overflow happen?

Reply:
I think when i added it not setting the menu to null made a stack overflow, but I think the comment can be removed